### PR TITLE
Modified patient id creation

### PR DESCRIPTION
**Pull request by wynnliam**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
New check in credentials for patient.

* **What is the current behavior?** (You can also link to an open issue here)
Currently just first name, last name, and last 4 digits of social security number for credentials when checking in.

* **What is the new behavior (if this is a feature change)?**
Adds first name, last name, email, and user code credentials for check in.

* **Other information**:
Updates UI to accommodate new credentials

### CODE REVIEW

File: src/app/check-in/check-in.component.ts [hunk 1]
```diff
@@ -13,96 +13,94 @@ import { ThsDataService } from '../services/ths-data.service';
...
+    if(this.isCredentialsValid()) {
+      sessionStorage.clear();
+      this.tsDataService.clearHistory();
+      this.tfiDataService.clearHistory();
+      this.thsDataService.clearHistory();
+      Utilities.setSessionStorage('patient-id', this.patientId);
+      Utilities.setSessionStorage('first-name', this.firstName);
+      Utilities.setSessionStorage('last-name', this.lastName);
+      Utilities.setSessionStorage('email', this.email);
+      console.log(this.patientId);
```
Comment (ElijahRW) [inline src/app/check-in/check-in.component.ts:34]: We don't need the console logs.

File: src/app/check-in/check-in.component.html [hunk 2]
```diff
@@ -25,28 +25,31 @@
...
     <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
-      <label><input id="lastfourssn" type="text" autofocus [(ngModel)]='lastFourSS' placeholder="Last 4 digits of SSN" > </label>
+      <label><input id="usercode" type="text" autofocus [(ngModel)]='patientId' placeholder="Usercode" > </label>
     </div>
+
   </div>
+
     <div class="row">
       <div class="col-sm-8 col-md-6 col-lg-4 col-lg-offset-4 col-md-offset-3 col-sm-offset-2">
         <div *ngIf="!authenticationFlag" class="failLogIn">One or more of the above fields is incomplete.</div>
```
Comment (ElijahRW) [inline src/app/check-in/check-in.component.html:55]: Can you change the button to not say `Log In` ?

File: src/app/check-in/check-in.component.ts [hunk 3]
```diff
@@ -13,96 +13,94 @@ import { ThsDataService } from '../services/ths-data.service';
...
+    } else {
+      this.authenticationFlag = false;
+      this.patientId = '';
+      console.log('failed log in ' + this.patientId);
+    }
+  }
 
-   constructor(private router: Router, private tsDataService: TsScreenerDataService, private tfiDataService: TfiDataService, private thsDataService: ThsDataService) {};
+  isCredentialsValid() : boolean {
+    return this.isEmailValid() && this.isNameValid() && this.isPatientIdValid();
```
Comment (ElijahRW) [inline src/app/check-in/check-in.component.ts:48]: This logic will fail if the user doesn't provide an email or a first or last name. We should be able to ONLY provide the user ID and we should still pass all checks.

**Review by ElijahRW — CHANGES_REQUESTED** [review]
Currently the following login fails that should pass:
* name: _eprw_
* lastname: _eprw_
* email: _eprw@gmail.com_
* id: _1234_

**Review by Liriaene — COMMENTED** [review]
+1 with Elijah's comments. Everything else looks good to me though.

Not going to leave an approval just yet however since I think if I approve it will allow a merge.

**Pull request closed without merging**